gdb/riscv-tdep.c

```diff
@@ -985,6 +985,8 @@ riscv_register_reggroup_p (struct gdbarch  *gdbarch, int regnum,
     {
       if (regnum < RISCV_FIRST_CSR_REGNUM || regnum == RISCV_PRIV_REGNUM)
 	return 1;
+      if (regnum >= RISCV_FIRST_CHERI_REGNUM && regnum <= RISCV_LAST_CHERI_REGNUM)
```

Comment:
I do wonder if the better fix is actually for the first 'if' here to be:

```
if (regnum < RISCV_FIRST_CSR_REGNUM || regnum > RISCV_LAST_CSR_REGNUM)
   return 1;
```

That is something I could upstream, or perhaps even:

```
if (regnum >= RISCV_FIRST_CSR_REGNUM && regnum <= RISCV_LAST_CSR_REGNUM)
    return riscv_is_regnum_a_named_csr (regnum);
return 1;
```

Reply:
This function is quite different upstream, so you can fix this however for now.  Your existing patch is fine.